### ISSUE TITLE
Missing instructions for reliable running the distributed OTP tests

### ISSUE THREAD

**Issue by frekw** [elixir-lang/elixir-lang.github.com#632]
Hi!

I was left scratching my head when trying to run the distributed `router_tests.exs` giving me the following error message:

```bash
Node 1:
$ iex --sname bar -S mix

Node 2:
$ elixir --sname foo -S mix test
  1) test route requests across nodes (KV.RouterTest)
     test/router_test.exs:5
     ** (EXIT from #PID<0.150.0>) an exception was raised:
         ** (BadFunctionError) expected a function, got: #Function<1.82587937/0 in KV.Router.route/4>
             :erlang.apply/2
             (elixir) lib/task/supervised.ex:74: Task.Supervised.do_apply/2
             (elixir) lib/task/supervised.ex:50: Task.Supervised.reply/3
             (stdlib) proc_lib.erl:240: :proc_lib.init_p_do_apply/3
```

After some googling I found that it was probably caused by different versions of the BEAM bytecode being used on the different nodes, since `elixir --sname foo -S mix test` will use a different path in the `_build` directory (which is my best guess, still pretty new to Elixir).

The solution was to spawn the node using `MIX_ENV` set to `test`, `$ MIX_ENV=test iex --sname bar -S mix`. I think it would be a helpful update to others doing the tutorial.

I could send a pull request, but I'm a bit weary about giving a good explanation as to **why** it's actually needed to set the right `MIX_ENV`.

**Comment by josevalim** [comment]
Thank you! Those guides are meant to be fully reviewed after 1.2 is out, so I will make sure to improve this.

**Comment by josevalim** [comment]
Fixed in upcoming 1.2 branch.
